### PR TITLE
OCP-28718 - Configurable timeout for mhc 

### PR DESCRIPTION
**Pull request by miyadav**
https://issues.redhat.com/browse/OCPCLOUD-774

V3 runner link - https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/92814/console

### CODE REVIEW

**Comment by miyadav** [comment]
@sunzhaohua2 @jhou1 PTAL

File: features/machine/machine_healthcheck.feature [hunk 1]
```diff
@@ -130,3 +130,40 @@ Feature: MachineHealthCheck Test Scenarios
...
+
+    And I wait up to 600 seconds for the steps to pass:
+    """
+     When I run the :logs admin command with:
+     | resource_name | <%= pod.name %>                |
+     | c             | machine-healthcheck-controller |
+    Then the output should contain:
+     | Ensuring a requeue happens in 15m0  |
+    """
+   And I ensure "mhc-<%= machine_set.name %>" machinehealthcheck is deleted after scenario
```
Comment (sunzhaohua2) [inline features/machine/machine_healthcheck.feature:166]: this line should be in line 153, in case the above step failed, this step will never be executed

Comment (miyadav) [reply features/machine/machine_healthcheck.feature]: Thank you , made the change  , looks good..

File: features/machine/machine_healthcheck.feature [hunk 2]
```diff
@@ -130,3 +130,40 @@ Feature: MachineHealthCheck Test Scenarios
...
+
+    Given I clone a machineset and name it "machineset-clone-28718"
+
+    # Create MHC with configurable node startup timeout
+    When I run oc create over "<%= BushSlicer::HOME %>/testdata/cloud/mhc/mhc_configurabletimeout.yaml" replacing paths:
+      | n                                                                                  | openshift-machine-api       |
+      | ["metadata"]["name"]                                                               | mhc-<%= machine_set.name %> |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]    | <%= machine_set.cluster %>  |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"] | <%= machine_set.name %>     |
+      | ["spec"]["nodeStartupTimeout" ]                                                    | 15m                         |
```
Comment (sunzhaohua2) [inline features/machine/machine_healthcheck.feature:152]: Add `Then the step should succeed`

**Comment by sunzhaohua2** [comment]
Added one comment, others looks good to me.

File: features/machine/machine_healthcheck.feature [hunk 3]
```diff
@@ -130,3 +130,41 @@ Feature: MachineHealthCheck Test Scenarios
...
+  # @author miyadav@redhat.com
+  # @case_id OCP-28718
+  @admin
+  @destructive
+  Scenario: [MHC] - Machine Node startup timeout should be configurable
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+    Then I use the "openshift-machine-api" project
+    And admin ensures machine number is restored after scenario
+
```
Comment (pruan-rht) [inline features/machine/machine_healthcheck.feature:143]: @miyadav small nitpick... extra empty line 143 & 145

File: features/machine/machine_healthcheck.feature [hunk 4]
```diff
@@ -130,3 +130,41 @@ Feature: MachineHealthCheck Test Scenarios
...
+    # Create MHC with configurable node startup timeout
+    When I run oc create over "<%= BushSlicer::HOME %>/testdata/cloud/mhc/mhc_configurabletimeout.yaml" replacing paths:
+      | n                                                                                  | openshift-machine-api       |
+      | ["metadata"]["name"]                                                               | mhc-<%= machine_set.name %> |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]    | <%= machine_set.cluster %>  |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"] | <%= machine_set.name %>     |
+      | ["spec"]["nodeStartupTimeout" ]                                                    | 15m                         |
+    Then the step should succeed
+    And I ensure "mhc-<%= machine_set.name %>" machinehealthcheck is deleted after scenario
+
```
Comment (pruan-rht) [inline features/machine/machine_healthcheck.feature:155]: extra empty line

File: features/machine/machine_healthcheck.feature [hunk 5]
```diff
@@ -130,3 +130,41 @@ Feature: MachineHealthCheck Test Scenarios
...
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]    | <%= machine_set.cluster %>  |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"] | <%= machine_set.name %>     |
+      | ["spec"]["nodeStartupTimeout" ]                                                    | 15m                         |
+    Then the step should succeed
+    And I ensure "mhc-<%= machine_set.name %>" machinehealthcheck is deleted after scenario
+    
+    Given I create the 'Ready' unhealthyCondition
+    Then a pod becomes ready with labels:
+     | api=clusterapi, k8s-app=controller |
+
```
Comment (pruan-rht) [inline features/machine/machine_healthcheck.feature:159]: extra empty line

File: features/machine/machine_healthcheck.feature (hunk 3 above)
Comment (miyadav) [reply features/machine/machine_healthcheck.feature:143]: thanks @pruan-rht  , will take care of these going forward.

**Pull request merged**